$MirOS: ports/emulators/qemu/stable/patches/patch-configure,v 1.2 2008/05/11 20:35:48 tg Exp $
--- configure.orig	Sun Sep  4 17:11:09 2005
+++ configure	Sun May 11 19:41:27 2008
@@ -21,8 +21,8 @@ prefix=""
 interp_prefix="/usr/gnemul/qemu-%M"
 static="no"
 cross_prefix=""
-cc="gcc"
-host_cc="gcc"
+cc="${CC:-cc}"
+host_cc="${CC:-cc}"
 ar="ar"
 make="make"
 strip="strip"
@@ -41,7 +41,7 @@ case "$cpu" in
   alpha)
     cpu="alpha"
   ;;
-  "Power Macintosh"|ppc|ppc64)
+  "Power Macintosh"|macppc|ppc|ppc64)
     cpu="powerpc"
   ;;
   mips)
@@ -103,13 +103,25 @@ if [ "$cpu" = "i386" -o "$cpu" = "x86_64
     kqemu="yes"
 fi
 ;;
+MirBSD)
+bsd="yes"
+openbsd="yes"
+oss="yes"
+if [ "$cpu" = "i386" ] ; then
+    kqemu="yes"
+fi
+;;
 NetBSD)
 bsd="yes"
 oss="yes"
 ;;
 OpenBSD)
 bsd="yes"
+openbsd="yes"
 oss="yes"
+if [ "$cpu" = "i386" -o "$cpu" = "x86_64" ] ; then
+    kqemu="yes"
+fi
 ;;
 Darwin)
 bsd="yes"
@@ -305,7 +317,7 @@ aa="no"
 `$sdl_config --static-libs | grep \\\-laa > /dev/null` && aa="yes"
 sdl_static_libs=`$sdl_config --static-libs`
 if [ "$aa" = "yes" ] ; then
-  sdl_static_libs="$sdl_static_libs `aalib-config --static-libs`"
+  sdl_static_libs="$sdl_static_libs `aalib-config --static-libs 2>/dev/null`"
 fi
 
 if $cc -o $TMPE `$sdl_config --cflags 2> /dev/null` $TMPC $sdl_static_libs 2> /dev/null; then
@@ -366,54 +378,13 @@ else
 if test -z "$prefix" ; then
     prefix="/usr/local"
 fi
-mandir="$prefix/share/man"
+mandir="$prefix/man"
 datadir="$prefix/share/qemu"
 docdir="$prefix/share/doc/qemu"
 bindir="$prefix/bin"
 fi
 
-# kqemu support
-if test $kqemu = "yes" ; then
-    # test if the source code is installed
-    if test '!' -f "kqemu/Makefile" ; then 
-        kqemu="no"
-    fi
-fi
-  
-# Linux specific kqemu configuration
-if test $kqemu = "yes" -a $linux = "yes" ; then
-# find the kernel path
-if test -z "$kernel_path" ; then
-kernel_version=`uname -r`
-kernel_path="/lib/modules/$kernel_version/build"
-if test '!' -d "$kernel_path/include" ; then 
-    kernel_path="/usr/src/linux"
-    if test '!' -d "$kernel_path/include" ; then 
-        echo "Could not find kernel includes in /lib/modules or /usr/src/linux - cannot build the kqemu module"
-        kqemu="no"
-    fi
-fi
-fi
 
-if test $kqemu = "yes" ; then
-
-# test that the kernel config is present
-if test '!' -f "$kernel_path/Makefile" ; then
-    echo "No Makefile file present in $kernel_path - kqemu cannot be built"
-    kqemu="no"
-fi    
-
-# find build system (2.6 or legacy)
-kbuild26="yes"
-if grep -q "PATCHLEVEL = 4" $kernel_path/Makefile ; then
-kbuild26="no"
-fi
-
-fi # kqemu
-
-fi # kqemu and linux
-
-
 echo "Install prefix    $prefix"
 echo "BIOS directory    $datadir"
 echo "binary directory  $bindir"
@@ -434,9 +405,7 @@ if test "$darwin" = "yes" ; then
     echo "Cocoa support     $cocoa"
 fi
 echo "SDL support       $sdl"
-if test "$sdl" != "no" ; then
-    echo "SDL static link   $sdl_static"
-fi
+echo "SDL static link   $sdl_static"
 echo "mingw32 support   $mingw32"
 echo "Adlib support     $adlib"
 echo -n "FMOD support      $fmod"
@@ -445,17 +414,6 @@ if test $fmod = "yes"; then
 fi
 echo ""
 echo "kqemu support     $kqemu"
-if test $kqemu = "yes" -a $linux = "yes" ; then
-echo ""
-echo "KQEMU Linux module configuration:"
-echo "kernel sources    $kernel_path"
-echo -n "kbuild type       "
-if test $kbuild26 = "yes"; then
-echo "2.6"
-else
-echo "2.4"
-fi
-fi
 
 if test $sdl_too_old = "yes"; then
 echo "-> Your SDL version is too old - please upgrade to have SDL support"
@@ -483,7 +441,6 @@ echo "CC=$cc" >> $config_mak
 if test "$have_gcc3_options" = "yes" ; then
   echo "HAVE_GCC3_OPTIONS=yes" >> $config_mak
 fi
-echo "HOST_CC=$host_cc" >> $config_mak
 echo "AR=$ar" >> $config_mak
 echo "STRIP=$strip -s -R .comment -R .note" >> $config_mak
 echo "CFLAGS=$CFLAGS" >> $config_mak
@@ -567,6 +524,9 @@ fi
 if test "$oss" = "yes" ; then
   echo "CONFIG_OSS=yes" >> $config_mak
   echo "#define CONFIG_OSS 1" >> $config_h
+  if test "$openbsd"="yes" ; then
+    echo "CONFIG_OSS_LIBRARY=yes" >> $config_mak
+  fi
 fi
 if test "$fmod" = "yes" ; then
   echo "CONFIG_FMOD=yes" >> $config_mak
@@ -583,12 +543,6 @@ echo "\"" >> $config_h
 
 if test $kqemu = "yes" ; then
   echo "CONFIG_KQEMU=yes" >> $config_mak
-  if test $linux = "yes" ; then
-    echo "KERNEL_PATH=$kernel_path" >> $config_mak
-    if test $kbuild26 = "yes" ; then
-      echo "CONFIG_KBUILD26=yes" >> $config_mak
-    fi
-  fi
 fi
 echo "SRC_PATH=$source_path" >> $config_mak
 echo "TARGET_DIRS=$target_list" >> $config_mak
@@ -600,6 +554,16 @@ if [ "$bsd" = "yes" ] ; then
   echo "#define _BSD 1" >> $config_h
 fi
 
+if [ "$openbsd" = "yes" ] ; then
+  echo "#ifndef ENOTSUP" >>$config_h
+  echo "#define ENOTSUP 4096" >> $config_h
+  echo "#endif" >>$config_h
+  echo "#ifndef PRIx64" >>$config_h
+  echo "#define PRIx64 \"llX\"" >> $config_h
+  echo "#endif" >>$config_h
+fi     
+
+
 for target in $target_list; do 
 
 target_dir="$target"
@@ -733,7 +700,7 @@ if test "$target_user_only" = "no"; then
         fi
         echo -n "SDL_CFLAGS=`$sdl_config --cflags`" >> $config_mak
         if [ "${aa}" = "yes" ] ; then
-            echo -n " `aalib-config --cflags`" >> $config_mak ;
+            echo -n " `aalib-config --cflags 2>/dev/null`" >> $config_mak ;
         fi
         echo "" >> $config_mak
     fi
